app/models/miq_report/seeding.rb

```diff
@@ -1,76 +1,72 @@
-      if typ == "report"
-        dir = REPORT_DIR
-        pattern = "*/*.yaml"
-        cond = {:rpt_type => 'Default', :template_type => [typ, nil]}
-      else
-        dir = COMPARE_DIR
-        pattern = "*.yaml"
-        cond = {:rpt_type => 'Default', :template_type => typ}
+      seed_files.each do |f|
+        seed_record(f, reports.delete(seed_filename(f)))
       end
 
-      where(cond).where.not(:filename => nil).each do |f|
-        unless File.exist?(File.join(dir, f.filename))
-          $log.info("#{typ.titleize}: file [#{f.filename}] has been deleted from disk, deleting from model")
-          f.destroy
-        end
-      end
+      if reports.any?
+        _log.info("Deleting the following MiqReport(s) as they no longer exist: #{reports.keys.sort.collect(&:inspect).join(", ")}")
 
-      Dir.glob(File.join(dir, pattern)).sort.each do |f|
-        sync_from_file(f, dir, typ)
+        # TODO: Can we make this a delete by getting rid of the dependent destroy on miq_report_result and using the purger?
+        MiqReport.destroy(reports.values.map(&:id))
       end
     end
 
-    def sync_from_file(filename, dir, typ)
-      yml = YAML.load_file(filename)
-      rpt = {}
-      column_names.each { |c| rpt[c.to_sym] = yml[c] }
-      rpt.delete(:id)
-      rpt[:name] = yml["menu_name"].strip
-      rpt[:rpt_group] = File.basename(File.dirname(filename)).split("_").last
-      rpt[:rpt_type] = "Default"
-      rpt[:filename] = filename.sub(dir + "/", "")
-      # DB and filesystem have different precision
-      # so calling round is done in order to eliminate the second fractions diff
-      # otherwise the comparison of the file time and the report time from db
-      # will always be different
-      rpt[:file_mtime] = File.mtime(filename).utc.round
-      rpt[:priority] = File.basename(filename).split("_").first.to_i
-      rpt[:template_type] = typ
-      rec = find_by_filename(rpt[:filename])
-
-      if rec
-        if rec.filename
-          _log.info("#{typ.titleize}: [#{rec.name}] file exists, synchronizing with model")
-          rec.update_attributes(rpt)
-          rec.save
-        end
-      else
-        _log.info("#{typ.titleize}: [#{rpt[:name]}] file has been added to disk, adding to model")
-        create(rpt)
+    # Used for seeding a specific report for test purposes
+    def seed_report(name)
+      path = seed_files.detect { |f| File.basename(f).include?(name) }
+      raise "report #{name.inspect} not found" if path.nil?
+
+      seed_record(path, MiqReport.find_by(:filename => seed_filename(path)))
+    end
+
+    private
+
+    def seed_record(path, report)
+      report ||= MiqReport.new
+
+      # DB and filesystem have different precision so calling round is done in
+      # order to eliminate the second fractions diff otherwise the comparison
+      # of the file time and the report time from db will always be different.
+      mtime = File.mtime(path).utc.round
+      report.file_mtime = mtime
```

Comment:
We should have stored the mtime of the file in the report record during the the initial seed

Reply:
I verified with @simaishi that, at least for downstream, when we rpm install a report, it will have the same mtime no matter when it's installed...i.e. rpm'ed reports store their mtime from build time inside the rpm.  For upstream, we expect people to use the same appliance, and those always have the same mtime.